src/dynamic_programming/fibonacci.rs

```diff
@@ -9,7 +9,7 @@ pub fn fibonacci(n: u32) -> u128 {
     // Use a and b to store the previous two values in the sequence
     let mut a = 0;
     let mut b = 1;
-    for i in 0..n {
+    for _ in 0..n {
```

Comment:
Let's take this out of the commit. We should address the compiler and clippy warnings in a separate PR.